src/classes/ADDR_AdminAcc_TEST.cls

```diff
@@ -0,0 +1,891 @@
+        
+        for (Account acc : mapAccIdAcc.values()) {
+            system.assertEquals('Direct Street Edit\r\nSecond Line', acc.BillingStreet);
+            system.assertEquals('Direct Street Edit\r\nSecond Line', acc.BillingStreet);
+            system.assertEquals(false, acc.is_Address_Override__c);
+            system.assertNotEquals(null, acc.Current_Address__c);
+        }
+ 
+        // verify the Address objects split street.
+        list<Address__c> listAddr = [select Id, MailingStreet__c, MailingStreet2__c, Formula_MailingStreetAddress__c, Default_Address__c from Address__c];
+        for (Address__c addr : listAddr) {
+            if (addr.Default_Address__c) {
+                system.assertEquals('Direct Street Edit', addr.MailingStreet__c);
+                system.assertEquals('Second Line', addr.MailingStreet2__c);
+                system.assertEquals('Direct Street Edit, Second Line', addr.Formula_MailingStreetAddress__c);
+            }
+        }
+    }
+    
+    /*********************************************************************************************************
+    @description Update accounts' billing address fields to empty cause clearing the default address  
+    verify:
+        existing Default Address updated to be non-default
+        no new Address created
+    **********************************************************************************************************/            
+    static testMethod void clearAddrExisting() {
+        UTIL_CustomSettingsFacade.getSettingsForTests(new Hierarchy_Settings__c(Accounts_Addresses_Enabled__c = 
+                                                                    UTIL_Describe.getAdminAccRecTypeID() + ';'));
+                
+        // this creates a default Address for each Account
+        UTIL_UnitTestData_TEST.AccsWithAddrs accsAddrs = UTIL_UnitTestData_TEST.createTestAccsAddrs(4, UTIL_Describe.getAdminAccRecTypeID());
+        UTIL_Debug.debug('****accsAddrs: ' + JSON.serializePretty(accsAddrs));
+        List<Address__c> addrs = accsAddrs.addrs;
+        
+        // clear addresses fields directly
+        for (Integer i = 0; i < accsAddrs.accs.size(); i++) {
+            accsAddrs.accs[i].BillingStreet = null;
+            accsAddrs.accs[i].BillingCity = null;
+            accsAddrs.accs[i].BillingState = null;
+            accsAddrs.accs[i].BillingPostalCode = null;
+            accsAddrs.accs[i].BillingCountry = null;
+            if (ADDR_Addresses_UTIL.isStateCountryPicklistsEnabled) {
+                accsAddrs.accs[i].put('BillingStateCode', null);
+                accsAddrs.accs[i].put('BillingCountryCode', null);
+            }             
+        }
+
+        ADDR_Addresses_TDTM.hasRunAddrTrigger = false;
+        Test.startTest();
+        update accsAddrs.accs;
+        Test.stopTest();
+        
+        // verify that no new address was created
+        addrs = [select Default_Address__c from Address__c];
```

Comment:
If the table in this query contains more than 50'000 records a runtime exception will be thrown. Consider either introducing a `WHERE` clause, or adding a `LIMIT` to contain the number of records returned.